schematic/schemas/json_schema_generator.py

```diff
@@ -0,0 +1,747 @@
+    # When the watched_property == conditional_value, the conditional_property will become required
+    for conditional_value in sorted(reverse_dependencies[property_display_name]):
+        if conditional_value in range_domain_map:
+            properties = sorted(range_domain_map[conditional_value])
+            # watched_property is the property such that when the
+            # watched_property == conditional_value
+            # the conditional_property becomes required
+            for watched_property in properties:
+                conditional_schema = {
+                    "if": {
+                        "properties": {
+                            watched_property: {"enum": sorted([conditional_value])}
+                        }
+                    },
+                    "then": {
+                        "properties": {conditional_property: {"not": {"type": "null"}}},
+                        "required": [conditional_property],
+                    },
+                }
+                json_schema.all_of.append(conditional_schema)
+
+
+def _set_property(  # pylint: disable=too-many-arguments
+    json_schema: JSONSchema,
+    name: str,
+    enum_list: list[str],
+    validation_rules: list[str],
+    is_required: bool,
+    description: str,
+) -> None:
+    """
+    Sets a property in the JSON schema. that is required by the schema
+
+    Arguments:
+        json_schema: The JSON Schema to modify
+        name: The name of the property in the JSON Schema to set
+        enum_list: A list of enums(valid values) for this property
+        validation_rules: The validation rules for the property
+        is_required: Whether or not the property is required
+        description: a description of the property
+    """
+    property_data = _get_property_data_from_validation_rules(validation_rules)
+    if enum_list:
+        if property_data.is_array:
+            schema_property = _create_enum_array_property(
+                name=name,
+                enum_list=enum_list,
+                is_required=is_required,
+                description=description,
+            )
+        else:
+            schema_property = _create_enum_property(
+                name=name,
+                enum_list=enum_list,
+                is_required=is_required,
+                description=description,
+            )
+
+    else:
+        if property_data.is_array:
+            schema_property = _create_array_property(
+                name=name,
+                property_data=property_data,
+                is_required=is_required,
+                description=description,
+            )
+        else:
+            schema_property = _create_simple_property(
+                name=name,
+                property_data=property_data,
+                is_required=is_required,
+                description=description,
+            )
+
+    json_schema.properties.update(schema_property)
+
+    if is_required:
+        json_schema.required += [name]
```

Comment:
This is a good candidate to be a method on the JSONSchema class. Usually it's bad practice to update the values of an argument that you pass into a function